src/request.cc

```diff
@@ -0,0 +1,80 @@
+#include "request_handler.h"
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <sstream>
+#include <vector>
+
+std::unique_ptr<Request> Request::Parse(const std::string& raw_request){
+    
+    std::unique_ptr<Request> request(new Request());
+
+    request->body_ = "";
+    request->raw_request_ = raw_request;
+
+    std::istringstream request_stream(raw_request);
+    std::string request_line;
+
+    //handle first line
+    if (std::getline(request_stream, request_line)) {
+        std::vector<std::string> tokens;
+        boost::algorithm::split(tokens, request_line, boost::algorithm::is_any_of(" \r"));
+
+        if (!tokens.empty() && tokens[0] == "GET") {
+            request->method_ = tokens[0];
+            request->uri_ = tokens[1];
+            request->version_ = tokens[2];
```

Comment:
Maybe you should check that tokens is of length 3 before accessing tokens[1] and tokens[2]?